src/internal.c

```diff
@@ -18097,12 +18098,24 @@ int ProcessReplyEx(WOLFSSL* ssl, int allowSocketErr)
 
 #if defined(WOLFSSL_DTLS) && defined(WOLFSSL_ASYNC_CRYPT)
     /* process any pending DTLS messages - this flow can happen with async */
-    if (ssl->dtls_rx_msg_list != NULL) {
-        ret = DtlsMsgDrain(ssl);
+    if (ssl->options.dtls && ssl->dtls_rx_msg_list != NULL) {
```

Comment:
Is it possible for `ssl->dtls_rx_msg_list` to be set without `ssl->options.dtls`? I feel like this should be an error if we have dtls on and a non-empty rx list.

Reply:
you are correct. It shouldn't be possible. The extra check does not harm, do you think I should remove it?

Reply:
If you want to check `ssl->options.dtls`, then I think `!ssl->options.dtls && ssl->dtls_rx_msg_list != NULL` should result in a log and an error. I don't know which error because I don't a wrong internal state error code (maybe add a new generic one?).

Reply:
ok, I'll remove the check for now.